Closes #17 - clean-up the code (#21)
holgerhagen/taskana-doc: docs/user-guide/configuration/taskana-jobs.md

|------------------------------------|-----------------------------------------------------------------|--------------|
| taskana.jobs.maxRetries | number of automatic retries after a job has failed        | 5            |
| taskana.jobs.batchSize  | upper bound of how many tasks can be processed by one job           | 45           |

## Example
**Setup**: In this example, the application starts on 5th April 2017 at 5pm. The scheduler has an initialStartDelay of 2 hours. The period of the scheduler is 3 hours. There is one Job active.This Job should be first run on 5th April at 6pm. The Jobs "runEvery" parameter is set to 2 hours.

**Result**: 
- 5th April 5pm: Application starts.
- 5th April 6pm: The Job becomes due. However, it isn't run yet.
- 5th April 7pm: Scheduler runs the first time. It runs the Job.
- 5th April 9pm: The Job becomes due. However, it isn't run yet.
- 5th Aril 10pm: Scheduler runs the second time. It runs the Job.
- 5th Aril 12pm: The Job becomes due. However, it isn't run yet.
- 6th April 1am: Scheduler runs the third time. It runs the Job.
- 6th Aril 3am: The Job becomes due. However, it isn't run yet.

> ## TaskCleanupJob and HistoryCleanupJob Configuration

Reviewer: For all jobs: Could you change the "Sample Value" to our actual default values? This gives a sample as well and gives even more info with the actual default. Also name the column default value pls

Author: What do you think about adding a "default value" column additionaly to the "sample value" column?

| Parameter                               | Description                                                                                                               | Sample Value         |
|-----------------------------------------|---------------------------------------------------------------------------------------------------------------------------|----------------------|

> | taskana.jobs.cleanup.task.runEvery           | period of time between the executions of the cleanup job on tasks (Duration in ISO 8601 format)                                                       | P2D                  |

Reviewer: I'm missing the "allCompletedSameParentBusiness" property for the TaskCleanupJob
-> taskana.jobs.cleanup.task.allCompletedSameParentBusiness

and also for simplehistory:
-> taskana.jobs.cleanup.history.simple.allCompletedSameParentBusiness

| taskana.jobs.cleanup.task.firstRunAt         | first time the cleanup job on tasks is run (DateTime n ISO 8601 format)                                                              | 2021-08-03T08:00:00Z |
| taskana.jobs.cleanup.task.minimumAge         | the completed task can be deleted by the cleanup only after this period of time or later  (Duration in ISO 8601 format)        | P10D                 |
| taskana.jobs.cleanup.history.simple.runEvery   | period of time between the executions of the cleanup job on history events (Duration in ISO 8601 format)                                                       | P7D                  |
| taskana.jobs.cleanup.history.simple.firstRunAt | first time the cleanup job on history events is run (DateTime n ISO 8601 format)                                                              | 2021-08-03T08:00:00Z |

> | taskana.jobs.cleanup.history.simple.minimumAge | the created history event can be deleted by the cleanup only after this period of time or later  (Duration in ISO 8601 format) | P60D                 |

Reviewer: ```suggestion
| taskana.jobs.cleanup.history.simple.minimumAge | a history event may only be deleted by the cleanup after this period of time or later  (Duration in ISO 8601 format) | P60D                 |
```

| taskana.jobs.cleanup.history.simple.batchSize     | upper bound of how many history events can be processed by one history cleanup job | 45           |

## TaskUpdatePriorityJob Configuration

| Parameter                        | Description                                                           | Sample Value         |
|----------------------------------|-----------------------------------------------------------------------|----------------------|
| taskana.jobs.priority.task.batchSize   | upper bound of how many tasks can be processed by one TaskUpdatePriorityJob                | 70                  |
| taskana.jobs.priority.task.runEvery   | period of time between the executions of the TaskUpdatePriorityJob (Duration in ISO 8601 format)   | P2D                  |
| taskana.jobs.priority.task.firstRunAt | first time the job is executed (DateTime in ISO 8601 format)                 | 2021-08-03T08:00:00Z |

> | taskana.jobs.priority.task.active     | the job will only be executed if the flag is set to true | true                |

Reviewer: isn't this taskana.jobs.priority.task.enable? Same for the other jobs
Also, if this is set to false, it won't even be scheduled, which technically is a difference to not being executed (but scheduled and therefore having an entry in the database)
